packages/manager/modules/pci/src/projects/project/sidebar/sidebar.controller.js

```diff
@@ -75,19 +77,40 @@ export default class SidebarController {
       });
   }
 
+  static findFeatureToCheck() {
+    return MENU.reduce((features, item) => {
+      return [
+        ...features,
+        item.feature,
+        ...item.subitems.map((subitem) => subitem.feature),
+      ].filter((feature) => !!feature);
+    }, []);
+  }
+
   $onInit() {
-    this.MENU = MENU.filter(
-      ({ regions }) => isNil(regions) || this.coreConfig.isRegion(regions),
-    ).map((menu) => {
-      set(
-        menu,
-        'subitems',
-        menu.subitems.filter(
-          ({ regions }) => isNil(regions) || this.coreConfig.isRegion(regions),
-        ),
-      );
-      return menu;
-    });
+    const featuresName = this.constructor.findFeatureToCheck();
+
+    this.ovhFeatureFlipping
+      .checkFeatureAvailability(featuresName)
+      .then((features) => {
+        this.MENU = MENU.filter(
+          ({ regions, feature }) =>
+            (isNil(regions) || this.coreConfig.isRegion(regions)) &&
+            (!feature || features.isFeatureAvailable(feature)),
+        ).map((menu) => {
+          set(
+            menu,
+            'subitems',
+            menu.subitems.filter(
+              ({ regions, feature }) =>
+                (isNil(regions) || this.coreConfig.isRegion(regions)) &&
+                (!feature || features.isFeatureAvailable(feature)),
+            ),
+          );
+          return menu;
+        });
```

Comment:
NP: 
```suggestion
        const isItemAvailable = (regions, feature) => (isNil(regions) || this.coreConfig.isRegion(regions)) && (!feature || features.isFeatureAvailable(feature));
        this.MENU = MENU.filter(
          ({ regions, feature }) => isItemAvailable(regions, feature),
        ).map((menu) => {
          set(
            menu,
            'subitems',
            menu.subitems.filter(
              ({ regions, feature }) => isItemAvailable(regions, feature),
            ),
          );
          return menu;
        });
```

Reply:
I find it way more better/readable.
Thank you